### ISSUE TITLE
Support for train / valid split in pretraining

### ISSUE THREAD

**Issue by franp9am** [s3prl/s3prl#363]
Would it be possible to add support for train and validation loss (or other metrics) in pretraining?

It would include something like defining the metadata in configuration files, supporting another dataloader, going over the validation set at the end of every epoch, aggregating loss and logging (at least).

Probably relevant file / place https://github.com/s3prl/s3prl/blob/master/s3prl/pretrain/runner.py#L154  

Should I try to do it?

**Comment by andi611** [comment]
Hello!

That sounds nice, we will be supporting validation loss for pertaining in our next version. 
It has been implemented but is not yet ready to release,
because we have re-designed the codebase and we are still finalizing the docs and unit tests.
The date of release is estimated for 1~2 months.

If you are in a hurry, you can try to do it on the current version, 
and feel free to make a PR when you are done!

**Comment by franp9am** [comment]
Thanks Andi, if it is already implemented, then I will probably rather wait and not do a PR.
